cross_corr_fft.py

```diff
@@ -62,254 +48,161 @@ def get_pdiff(parr1, parr2, start, end):
     return np.mean(ave)
 
-
-def cross_corr_func(filename, if_double, version, if_plot, samp_f=fs, tar_f=freq, guess_x=gx, guess_y=gy, guess_z=gz):
-    global fs, freq, gx, gy, gz, ax
+def cross_corr_func(filename, if_double, version, if_plot, samp_f=fs, tar_f=freq, guess_x=guess[0], guess_y=guess[1], guess_z=guess[2]):
+    global fs, freq, guess
     fs = samp_f
     freq = tar_f
     filepath = filename
-    gx = guess_x
-    gy = guess_y
-    gz = guess_z
+    guess = (guess_x, guess_y, guess_z)
 
     ccwha = []
     downva = []
-    xa = []
-    ya = []
-    za = []
+    actual = []
     count = 0
 
-    if version > 0:
-        for i in range(version):
-            data1, data2, data3, data4 = read_data(filepath.replace(".csv", "("+str(i+1)+").csv"))
-
-            if if_double:
-                data1a = [data1[0:int(len(data1)/2)], data1[int(len(data1)/2):len(data1)]]
-                data2a = [data2[0:int(len(data1)/2)], data2[int(len(data1)/2):len(data1)]]
-                data3a = [data3[0:int(len(data1)/2)], data3[int(len(data1)/2):len(data1)]]
-                data4a = [data4[0:int(len(data1)/2)], data4[int(len(data1)/2):len(data1)]]
-                double = 2
-            else:
-                data1a = [data1]
-                data2a = [data2]
-                data3a = [data3]
-                data4a = [data4]
-                double = 1
-
-            for j in range(double):
-                data1 = data1a[j]
-                data2 = data2a[j]
-                data3 = data3a[j]
-                data4 = data4a[j]
-                pdiff12, pdiff13, pdiff34 = data_to_pdiff(data1, data2, data3, data4)
-
-                if (pdiff12 is not None) and (pdiff13 is not None) and (pdiff34 is not None):
-                    count = count + 1
-                    diff12 = pdiff12/2/np.pi*vsound/freq
-                    diff13 = pdiff13/2/np.pi*vsound/freq
-                    diff34 = pdiff34/2/np.pi*vsound/freq
-
-                    guess = (gx, gy, gz)
-                    x, y, z = solver(guess, diff12, diff13, diff34)
-                    xa.append(x)
-                    ya.append(y)
-                    za.append(z)
-                    print("initial guess", guess)
-                    print("x, y, z", x, y, z)
-
-                    # calculate angle
-                    ccwh = np.arctan2(y, x)
-                    ccwha.append(ccwh)
-                    print("horizontal angle", ccwh/np.pi*180)
-                    downv = np.arctan2(-z, np.sqrt(x**2 + y**2))
-                    downva.append(downv)
-                    print("vertical downward angle", downv/np.pi*180, "\n")
-
-                    # compare solver result with pdiff from data
-                    pd12, pd13, pd34 = check_angle(ccwh, downv)
-                    # print("checked pd", pd12, pd13, pd34, "\n")
-                    # print("pdiff_12", pdiff12, "pdiff_13", pdiff13, "pdiff_34", pdiff34, "\n")
+    for i in range(version):
```

Comment:
i forgot to write the else condition for version>0 here, if version == 0, `raw_data = read_data(filepath)` instead

Reply:
How does this work? Specifically, if version == 0, then how is it supposed to enter the for loop where all the calculations and such are being performed?

Reply:
Yes it should still enter the loop, but it would not be a for loop since there is only one file, and the filename does not need to be modified. so it will still execute the lines starting `data = [split_data(d) if if_double else [d] for d in raw_data]`